### PR TITLE
[VL] Add write IO metrics for WriteFiles

### PR DESCRIPTION
**Pull request by Yohahaha**

### CODE REVIEW

**Comment by Yohahaha** [comment]
depends on #7002

File: gluten-data/src/main/scala/org/apache/gluten/metrics/MetricsUtil.scala [hunk 1]
```diff
@@ -182,6 +183,7 @@ object MetricsUtil extends Logging {
       ioWaitTime,
       preloadSplits,
       physicalWrittenBytes,
+      writeIOTime,
```
Comment (zhztheplayer) [inline gluten-data/src/main/scala/org/apache/gluten/metrics/MetricsUtil.scala:186]: Perhaps you missed the merge code?

**Comment by Yohahaha** [comment]
cc @PHILO-HE @liujiayi771 thank you!

**Comment by Yohahaha** [comment]
@JkSelf

**Review by PHILO-HE — APPROVED** [review]

**Comment by zhztheplayer** [comment]
@Yohahaha Can you confirm, there is a list for merging metric values https://github.com/Yohahaha/gluten/blob/f58163433b83453b772d76a158503bb1a77e28ad/gluten-data/src/main/scala/org/apache/gluten/metrics/MetricsUtil.scala#L129-L153 and whether we should append some code into that list.

cc @PHILO-HE

**Comment by Yohahaha** [comment]
> @Yohahaha Can you confirm, there is a list for merging metric values https://github.com/Yohahaha/gluten/blob/f58163433b83453b772d76a158503bb1a77e28ad/gluten-data/src/main/scala/org/apache/gluten/metrics/MetricsUtil.scala#L129-L153 and whether we should append some code into that list.

I see, TableWriter count `physicalWrittenBytes`, `writeIONanos`, `numWrittenFiles`, but only `numWrittenFiles` is merged.

why we need merge those metrics?

**Comment by zhztheplayer** [comment]
Sorry, I am not aside a PC now, so I can't actually reason about the code, which looks too magical to me by the way. 

But what's clear is only input and output metrics were being handled separately when the code was [firstly written](https://github.com/Yohahaha/gluten/commit/7996326ddf6e1c8441d3184a8e39036d501ea2ee) .

>  TableWriter count physicalWrittenBytes, writeIONanos, numWrittenFiles, but only numWrittenFiles is merged

After all, I think we should make these 4 metrics consistent. Either merge them all, or none of them. And seems to be safe to merge them all by taking one look. cc @JkSelf

**Pull request merged**